Content/Tools/Spawners/BuffSpawner.cs

```diff
@@ -195,6 +197,10 @@ public override void Draw(SpriteBatch spriteBatch)
 				spriteBatch.Draw(tex, Main.MouseScreen + Vector2.One * 8, new Rectangle(0, 0, tex.Width, tex.Height), Color.White * alpha, 0, default, 1, 0, 0);
 			}
 
+			// Set name here to receive game language selection changes in real time
+			// This is a bit of a hack, but it works
+			durationEditor.name = BuffSpawner.GetText("DurationEditor");

```

Comment:
I wonder if perhaps a more sensible solution is to have a watchdog-type system that will re-initialize all of the UI when the language updates? This is likely an infrequent event so would probably not be a performance issue, and would allow the behavior to be centralized and automatic so there is no pitfall of forgetting to add this when implementing a new tool for example.

Reply:
We can listen to an event in I believe `LanguageManager` to know when the language changes; flagging for initialization is trivial there.

Reply:
Sounds like a plan. Lets create an issue to look into this